### PR TITLE
Add the evaluation for text-to-sql task (spider dataset)

### PR DESCRIPTION
**Pull request by shuaichenchang**
Contains 2 evaluation metrics for SQL prediction:
1) Exact Set Match Accuracy (EM)
2) Execution Accuracy (EX)
The definition can be found in https://arxiv.org/pdf/1809.08887.pdf 
The evaluation scripts are adapted from https://github.com/taoyds/test-suite-sql-eval

### CODE REVIEW

**Comment by pfliu-nlp** [comment]
Overall, it looks very nice!! thanks for your contribution!
We can have a meeting to discuss more details besides the above general comments.
Also, the file `question.json` is too large, I will put it in S3 so that we can download and cache it first.

File: explainaboard/metrics/auxiliary/sql_evaluation/exec_eval.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
+import asyncio
```
Comment (odashi) [inline explainaboard/metrics/auxiliary/sql_evaluation/exec_eval.py:1]: It looks these files are borrowed from somewhere we don't recognize, but there are no explanation about borrowing. Please apply following changes:

- Move `explainaboard/metrics/auxiliary` to `explainaboard/contrib` to indicate this is a third-party.
- Check if the original license is compatible with the MIT license. If so, add the original licensing notes to this folder. If not, we can not borrow the code from such source.

File: explainaboard/metrics/sql_em_ex.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    table_path: str = ""
+    etype: str = "exec"
+
+    def to_metric(self):
+        return SQLEx(self)
+
+
+class SQLEx(Metric):
+    """
+    Calculate exact set match (Em) and execuation (Ex) accuracy,
```
Comment (odashi) [inline explainaboard/metrics/sql_em_ex.py:59]:
```suggestion
    Calculate exact set match (Em) and execution (Ex) accuracy,
```

File: explainaboard/metrics/sql_em_ex_test.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+"""Tests for explainaboard.metrics.sql_em_ex"""
```
Comment (odashi) [inline explainaboard/metrics/sql_em_ex_test.py:1]: Could you add unit tests for `calc_stats_from_data`?

File: explainaboard/processors/text_to_sql.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        """
+        return data_point["true_sql"]
+
+    def _get_predicted_label(self, data_point):
+        """
+        Get the predicted label from a data point. Overloaded from parent class.
+        :param data_point: the data point under consideration
+        :return: the predicted label for the output
+        """
+        return data_point["predicted_sql"]
```
Comment (odashi) [inline explainaboard/processors/text_to_sql.py]: These functions are private and not used in this class.

```suggestion
```

File: integration_tests/metric_test.py [hunk 5]
```diff
@@ -260,3 +261,49 @@ def test_qa_metrics(self):
             2,
             "almost equal",
         )
+
+    @unittest.skip(reason="dependency libs are not installed")
```
Comment (odashi) [inline integration_tests/metric_test.py:265]: Could we re-enable all skipped tests? In general, invalidated tests are not very useful to check consistency of the software.

If we really need to disable these tests, please add following things:

- which library is required to invoke this test
- why the library is not added to the dependency

File: integration_tests/metric_test.py [hunk 6]
```diff
@@ -260,3 +261,49 @@ def test_qa_metrics(self):
...
+    def test_sql_em(self):
+        # python -m unittest integration_tests.metric_test.MetricTest.test_sql_em
+        metric = explainaboard.metrics.sql_em_ex.SQLEmConfig(
+            name="SQLEm",
+            db_dir="https://expressai-xlab.s3.amazonaws.com/large_data/database",
```
Comment (odashi) [inline integration_tests/metric_test.py:270]: This is considered as a personal storage. We also have a Cloud Storage to store large data related to our software. @pfliu-nlp and @neubig have write access to it.

**Comment by pfliu-nlp** [comment]
@odashi This PR is still modified.  @shuaichenchang could you make this PR  a draft version?

**Pull request closed without merging**